### PR TITLE
[eclipse/xtext#1777] ported more xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1777] ported more xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/GeneratorNodeProcessor.java [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+	 * 
+	 * @author Sven Efftinge - Initial contribution and API
+	 */
+	public static class CompletableTraceRegion extends AbstractStatefulTraceRegion {
+		public static class CompletableTextRegion implements ITextRegionWithLineInformation {
+			private ITextRegionWithLineInformation delegate;
+
+			@Delegate
+			public ITextRegionWithLineInformation getDelegate() {
+				if ((delegate == null)) {
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/GeneratorNodeProcessor.java:223]: Double parentheses

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TemplateNode.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+				append(object);
+			} finally {
+				currentParent = before;
+			}
+		} else {
+			append(object);
+		}
+	}
+
+	private Splitter lineSplitter = Splitter.on(Pattern.compile("\\R"));
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TemplateNode.java:55]: can be static

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TemplateNode.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+				String segment = iter.next();
+				nodeFactory.append(currentParent, segment);
+				if (iter.hasNext()) {
+					newLine();
+				}
+			}
+		}
+	}
+
+	protected Iterable<IGeneratorNode> leafsBackwards(IGeneratorNode it) {
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TemplateNode.java:79]: unused? Maybe remove this method.

Comment (cdietrich) [reply org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TemplateNode.java:79]: deprecate and mark for removal?

**Pull request merged**